Addons/Fbx.cs

```diff
@@ -49,18 +64,35 @@ static Fbx()
 
         static void TryLoadFbxSupport()
         {
-            if (s_FbxIsLoaded)
+            if (FbxExporterAssembly == null)
+            {
                 return;
-            ModelExporter.RegisterMeshCallback<ProBuilderMesh>(GetMeshForComponent, true);
+            }
+
+            var modelExporter = FbxExporterAssembly.GetType("UnityEditor.Formats.Fbx.Exporter.ModelExporter");
+            var registerMeshCallback = modelExporter.GetMethods(BindingFlags.NonPublic | BindingFlags.Static).Where(x => x.Name == "RegisterMeshCallback").First(x => x.ContainsGenericParameters);
+            registerMeshCallback = registerMeshCallback.MakeGenericMethod(typeof(ProBuilderMesh));
+
+            var getMeshForComponent = FbxExporterAssembly.GetTypes()
+               .Where(t => t.BaseType == typeof(MulticastDelegate) && t.Name.StartsWith("GetMeshForComponent"))
+               .First(t => t.ContainsGenericParameters);
+            
+            getMeshForComponent = getMeshForComponent.MakeGenericType(typeof(ProBuilderMesh));
+            var meshDelegate = Delegate.CreateDelegate(getMeshForComponent, typeof(Fbx).GetMethod("GetMeshForComponent", BindingFlags.NonPublic | BindingFlags.Static));
+
+            registerMeshCallback.Invoke(null, new object[] { meshDelegate, true });
+            
             m_FbxOptions.quads = ProBuilderSettings.Get<bool>("Export::m_FbxQuads", SettingsScope.User, true);
-            s_FbxIsLoaded = true;
         }
-
-        static bool GetMeshForComponent(ModelExporter exporter, ProBuilderMesh pmesh, FbxNode node)
+        
+        internal static bool GetMeshForComponent(object exporter, ProBuilderMesh pmesh, object node)
         {
             Mesh mesh = new Mesh();
             MeshUtility.Compile(pmesh, mesh, m_FbxOptions.quads ? MeshTopology.Quads : MeshTopology.Triangles);
-            exporter.ExportMesh(mesh, node, pmesh.GetComponent<MeshRenderer>().sharedMaterials);
+
+            var exportMeshMethod = exporter.GetType().GetMethod("ExportMesh", BindingFlags.NonPublic | BindingFlags.Instance, null, new Type[] { typeof(Mesh), node.GetType(), typeof(Material[]) }, null);
```

Comment:
Let's write a comment about how we're calling `exporter.ExportMesh(mesh, node, pmesh.GetComponent<MeshRenderer>().sharedMaterials)` since that's a bit hidden by the reflection (ha!)

Maybe also put `pmesh.GetComponent<MeshRenderer>().sharedMaterials` into their own local variable since that's a bit of a mouthful.